code/datums/spells/lordspells.dm

```diff
@@ -0,0 +1,93 @@
+	amt_paralysis = 1
+	amt_dam_fire = 15
+	amt_dam_brute = 15
+	amt_dam_oxy = 15
+	sound="sound/magic/Necrolord_Soulflare_Cast.ogg"
+
+/obj/effect/proc_holder/spell/targeted/inflict_handler/soulflare/cast(list/targets, mob/user = usr)
+	var/obj/effect/proc_holder/spell/targeted/trigger/soulflare/SF = locate(/obj/effect/proc_holder/spell/targeted/trigger/soulflare, user.mob_spell_list)
+	var/mob/living/carbon/target = targets[1]
+	if(target.health <= 0)
+		if(!target.stat & DEAD)
+			target.adjustOxyLoss(500)
+			user << "<span class='notice'>You've successfully killed [target], refunding your spell and decreasing it's cooldown permanently.</span>"
+			user << 'sound/magic/Necrolord_Soulflare_Crit.ogg'
+			if(SF.charge_max >= 61)
+				SF.soulflarecm -= 10
+				SF.charge_max -= 10
+			SF.charge_counter = SF.soulflarecm
+		else
+			user << "<span class='warning'>[target] is already dead!</span>"
+			SF.charge_counter = SF.soulflarecm
+	..()
+
+/obj/effect/proc_holder/spell/targeted/explodecorpse
+	name = "Corpse Explosion"
+	desc = "Explodes a corpse in a very, very big and pretty explosion. The explosion is 9x9 centered on the target, so make sure to maintain distance when you cast it. Who needs maxcaps when you can just go green and recycle?"
+	school = "transmutation"
+	charge_max = 200
+	clothes_req = 1
+	invocation = "BO'NES T'O BO'MS"
+	invocation_type = "shout"
+	cooldown_min = 10
+	centcom_cancast = 0
+	sound="sound/magic/Necrolord_Soulflare_Cast.ogg"
+
+	action_icon_state = "raisedead"
+
+/obj/effect/proc_holder/spell/targeted/explodecorpse/cast(list/targets, mob/user = usr)
+	..()
+	var/mob/living/carbon/target = targets[1]
+	if(!target)
+		return
+	if(target.stat & DEAD)
+		message_admins("[user] casted corpse explosion on [target]")
+		explosion(target,1,2,5)
+		user << "<font color=purple><b>You redirect an absurd amount of energy into [target]'s corpse, causing it to violently explode!</font>"
+	else
+		user << "<span class='warning'>[target] isn't a dead corpse!</span>"
+		charge_counter = 60
+
+/obj/effect/proc_holder/spell/self/soulsplit
+	name = "Soulsplit"
+	desc = "Enter a wraith-like form, traveling at very high speeds and moving through objects. However, maintaining this form requires you to be at full health to maintain concentration!"
+	school = "transmutation"
+	charge_max = 300
+	clothes_req = 1
+	centcom_cancast = 0
+	invocation = "TRAVEL ME BONES"
+	invocation_type = "shout"
+	cooldown_min = 150
+
+	action_icon_state = "soulsplit"
+
+/obj/effect/proc_holder/spell/self/soulsplit/cast(list/targets, mob/living/user = usr)
+	if(user.health >= 100)
+		user << "<font color=purple><b>You enter your wraith form, leaving you vulnerable yet very manoeuvrable.</font>"
+		user.incorporeal_move = 2
+		spawn(35)
+			user << "<span class='warning'>Soulsplit wears off!</span>"
+			user.incorporeal_move = 0
+	else
+		user << "<span class='warning'>You cannot concentrate on casting soulsplit while injured!</span>"
+		charge_counter = 300
```

Comment:
I think you should go `charge_counter = init(charge_counter)` or `= charge_max` as that seems like its purpose